### PR TITLE
Include date in winstore Python cache key to limit cache age

### PR DESCRIPTION
**Pull request by rmartin16**
## Changes
- While GitHub will evict caches after not being used for 7 days, the cache will exist in perpetuity as long as it is regularly referenced.
- To ensure the Windows Store Python that's installed stays up to date, the current year and month is added to the cache key so they cache is recreated each month.
- Additionally, since this the first CI run since adding the test for RTD Builds, it is now clear the test will always fail for PRs from forks so I modified the test to only run for branches on this repo.

## Notes
- Caches with the year and date in the name can be seen [here](https://github.com/beeware/.github/actions/caches).
- GitHub does provide an API to delete caches....but it seems especially over-engineered to periodically check the age of these caches and delete them manually.

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -24,6 +24,8 @@ jobs:
   test-docs-build-verify:
     name: Test Verify Docs Build
     needs: pre-commit
+    # inheriting secrets does not work for forked repos
+    if: ${{ github.event.pull_request.head.repo.owner == 'beeware' }}
```
Comment (freakboy3742) [inline .github/workflows/ci.yml:28]: Is this an intentional inclusion? It doesn't seem related to the caching problem...

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
The cache change makes sense; the extra bit about inheriting secrets doesn't seem to be related, though. I *think* it makes sense (making sure that forked repos won't try to verify the RTD build, since they won't have credentials to do so); but I wanted to make sure the inclusion was intentional.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (rmartin16) [reply .github/workflows/ci.yml:28]: Apologies; I should have explicitly mentioned this.

It is intentional; the [first run](https://github.com/beeware/.github/actions/runs/4756507186) of CI failed because the `RTD_API_TOKEN` secret did not resolve. I considered removing the test altogether but this seemed like a better middle ground to ensure CI passes for PRs from forked repos going forward while the test can test at least run sometimes.

P.S. - when I originally added this test, I moved the PR from my fork to this repo; at the time, I was hoping that future PRs from forks wouldn't hit the same problem....I was wrong...

**Review by freakboy3742 — APPROVED** [review]
Ah - that makes sense - thanks for the clarification.
